dhcpv4/dhcpv4.go

```diff
@@ -730,6 +730,22 @@ func (d *DHCPv4) IPAddressRebindingTime(def time.Duration) time.Duration {
 	return time.Duration(dur)
 }
 
+// IPv6OnlyPreferred returns the V6ONLY_WAIT duration, and a boolean
+// indicating whether this option was present.
+//
+// The IPv6-Only Preferred option is described by RFC 8925, Section 3.1.
+func (d *DHCPv4) IPv6OnlyPreferred() (time.Duration, bool) {
```

Comment:
Will this ever return 0, true? If not shouldn't we simply return time.Duration as in `IPAddressRebindingTime`?

Reply:
Yes it makes a difference. Setting the option in the response with value 0, is a signal to the client that it can use IPv6-only for an unlimited time (i.e. never check back for DHCPv4)

Two other possibilities I can think of:
1. Return some sentinel value for -1 meaning "this option was not set" (that's ugly)
2. return (value, error) with a non-nil error if the option is not supplied. That's how `DHCPv4.MaxMessageSize()` works.

Reply:
> Yes it will, and it makes a difference. Setting the option in the response with value 0, is a signal to the client that it can use IPv6-only for an unlimited time

From the RFC I get something different but what matters is that having a zero or not having it makes a difference:

```
If the IPv6-Only Preferred option returned by the server contains a value greater than or equal to MIN_V6ONLY_WAIT, the client SHOULD set the V6ONLY_WAIT timer to that value. Otherwise, the client SHOULD set the V6ONLY_WAIT timer to MIN_V6ONLY_WAIT.
```

Reply:
Yes, on re-reading you are correct. I note that section 3.4 says that V6ONLY_WAIT has a "default" of 1800 seconds; and yet it says twice (in 3.1 and 3.3) that if the server has no explicit configured value then it *must* send a value of zero, and if the client receives a value less than MIN_V6ONLY_WAIT then it should use MIN_V6ONLY_WAIT. So I cannot see any circumstance in which this default value of 1800 would be used.

It might be reasonable to clip the returned value to MIN_V6ONLY_WAIT - although it's a "should" not a "must".

In any case: do you prefer "ok bool" or "err error" for the presence indication?

Reply:
> In any case: do you prefer "ok bool" or "err error" for the presence indication?

I have no preference.

Reply:
I don't even need it for coredhcp server; only a client needs to look for the presence of the option (the client never sends it to server, only includes it in parameter request list).

As a client, I would probably not care about the difference between a missing option versus a malformed option (and if I did, I would need a sentinel error for "missing").

I'll stick with bool then, but I will change #525 for autoconfigure option to be consistent.